OIPA/iati/parser/tests/test_2_03_parser.py

```diff
@@ -2095,3 +2095,103 @@ def test_activity_result_indicator_document_link_document_category(self):
+        xml_file_attrs = {
+            "generated-datetime": datetime.datetime.now().isoformat(),
+            "version": '2.03',
+        }
+        self.iati_203_XML_file = E("iati-activities", **xml_file_attrs)
+
+        dummy_source = synchroniser_factory.DatasetFactory.create()
+
+        self.parser_203 = ParseManager(
+            dataset=dummy_source,
+            root=self.iati_203_XML_file,
+        ).get_parser()
+
+        self.document_link = iati_factory.DocumentLinkFactory. \
+            create(url='http://someuri.com')
+
+        self.parser_203.register_model('DocumentLink',
+                                       self.document_link)
+
+    def test_activity_result_indicator_document_link_language(self):
+        '''
+        Test if language attribute in <document_link_language> XML element is
+        correctly saved.
+        '''
+
+        # case 1: 'code' is missing
+
+        indicator_document_link_language_attr = {
+            # "code": 'en'
+
+        }
+        indicator_document_link_language_XML_element = E(
+            'document-link-language',
+            **indicator_document_link_language_attr
+        )
+
+        try:
+            self.date = self.parser_203 \
+                 .iati_activities__iati_activity__result__indicator__document_link__language(  # NOQA: E501
+
+                indicator_document_link_language_XML_element
+
+            )
+        except RequiredFieldError as inst:
+            self.assertEqual(inst.field, 'code')
+            self.assertEqual(inst.message, 'required attribute missing')
+
+        # case 2: 'language' is not found
+        indicator_document_link_language_attr = {
+
+            "code": 'ab'
+
+        }
+        indicator_document_link_language_XML_element = E(
+            'document-link-language',
+            **indicator_document_link_language_attr
+        )
+        try:
+            self.parser_203.iati_activities__iati_activity__result__indicator__document_link__language(  # NOQA: E501
+                indicator_document_link_language_XML_element
+            )
+        except FieldValidationError as inst:
+            self.assertEqual(inst.field, 'code')
+            self.assertEqual(inst.message,
+                             'not found on the accompanying code list')
+
+        # all is good
+        language = codelist_factory.LanguageFactory()  # dummy language object
+        indicator_document_link_language_attr = {
+
+            "code": language.code
+
+        }
+        indicator_document_link_language_XML_element = E(
+            'document-link-language',
+            **indicator_document_link_language_attr
+        )
+        self.parser_203\
+            .iati_activities__iati_activity__result__indicator__document_link__language(  # NOQA: E501
+                indicator_document_link_language_XML_element
+            )
+
+        # Let's test language is saved
+
+        document_link = self.parser_203.get_model('DocumentLink')
```

Comment:
It's just a small thing, but I think you should actually use `self.document_link` object that you created in `setUp()` method here